### PR TITLE
Add infrastructure ConfigValidator for checking VPC IDs

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area ops-productivity
/area quality
/kind bug
/platform aws

**What this PR does / why we need it**:
* Adds an implementation of the `infrastructure.ConfigValidator` for checking if a VPC ID exists, has correct VPC attribute values, and have an internet gateway attached. See https://github.com/gardener/gardener/pull/4547.
* Vendors g/g v1.30.0-dev (in a separate commit).
* Adds unit tests for the `infrastructure.ConfigValidator` implementation (in a separate commit).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
* Based on https://github.com/gardener/gardener/pull/4547 so in Draft until this PR is merged and could be properly vendored, otherwise ready for review. 
* Similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/312

**Release note**:

```feature operator
VPC IDs in infrastructure config are now checked if they exist, have correct VPC attribute values, and have an internet gateway attached, and if not the issue is reported as `ERR_CONFIGURATION_PROBLEM` with a clear error message.
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
/hold
as there is already ongoing release validation

File: pkg/aws/client/client.go [hunk 1]
```diff
@@ -427,11 +420,15 @@ func (c *Client) DeleteSecurityGroup(ctx context.Context, id string) error {
 	return ignoreNotFound(err)
 }
 
-func ignoreNotFound(err error) error {
-	if err == nil {
-		return nil
+func IsNotFoundError(err error) bool {
```
Comment (rfranzke) [inline pkg/aws/client/client.go:423]: Nit: doc string missing

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
Let's reopen this PR or have a new one and have it in after the release of v1.28.0 (as we did a revert with 8f00f94).

**Pull request merged**